doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -198,6 +198,35 @@ system.
+<!-- have one or more corresponding executions in inclusion in -->
+<!-- [`mz_statement_execution_history`](#mz_statement_execution_history). -->
+
+<!-- | Field         | Type                         | Meaning                                                                                                                           | -->
+<!-- |---------------|------------------------------|-----------------------------------------------------------------------------------------------------------------------------------| -->
+<!-- | `id`          | [`uuid`]                     | The globally ID of the prepared statement.                                                                                        | -->
+<!-- | `session_id`  | [`uuid`]                     | The globally unique ID of the session that prepared the statement. Corresponds to [`mz_session_history.id`](#mz_session_history). | -->
+<!-- | `name`        | [`text`]                     | The name of the prepared statement (the default prepared statement's name is the empty string).                                   | -->
+<!-- | `sql`         | [`text`]                     | The SQL text of the prepared statement.                                                                                           | -->
+<!-- | `prepared_at` | [`timestamp with time zone`] | The time at which the statement was prepared.                                                                                     | -->
+
+<!-- ### `mz_session_history` -->
+
+<!-- The `mz_session_history` table contains all the sessions that have -->
+<!-- been established in the last 30 days, or (even if older) that are -->
+<!-- referenced from -->
+<!-- [`mz_prepared_statement_history`](#mz_prepared_statement_history). -->
+
+<!-- | Field          | Type                         | Meaning                                                                                                                           | -->
+<!-- |----------------|------------------------------|-----------------------------------------------------------------------------------------------------------------------------------| -->
+<!-- | `id`           | [`uuid`]                     | The globally unique ID of this history entry. Does **not** correspond to [`mz_sessions.id`](#mz_sessions), which can be recycled. | -->
+<!-- | `connected_at` | [`timestamp with time zone`] | The time at which the session was established.                                                                                    | -->
+<!-- | `application`  | [`text`]                     | The `application` session metadata field.                                                                                         | -->
+<!-- | `session_user` | [`text`]                     | The name of the user for wish the session was established.                                                                        | -->
```

Comment:
Will change it to `authenticated_user`, sure.

As for denormalization: this is a "history" table, in which case I think it makes sense to denormalize, because we are trying to record the state of the world at the time the event happened. Similar to `mz_audit_events`.